src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const stateHistory = [];
+
+  for (let index = 0; index < actions.length; index++) {
+    const action = actions[index];
+    const type = action.type;
+
+    switch (type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...action.extraData };
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (let i = 0; i < action.keysToRemove.length; i++) {
+          delete stateCopy[action.keysToRemove[i]];
+        }
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        const keys = Object.keys(stateCopy);
+
+        for (let i = 0; i < keys.length; i++) {
+          delete stateCopy[keys[i]];
+        }
+        stateHistory.push({});
```

Comment:
Instead of pushing a new empty object `{}` directly, you should assign an empty object to `stateCopy` and then push `stateCopy` to `stateHistory` after the `switch` block to maintain consistency and follow the DRY (Don't Repeat Yourself) principle.